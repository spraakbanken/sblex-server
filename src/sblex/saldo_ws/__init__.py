from sblex.saldo_ws import routes, server
from sblex.sblex_server import deps, responses, tasks, templating
from sblex.sblex_server.schemas import base

__all__ = ["deps", "server", "responses", "routes", "base", "tasks", "templating"]
