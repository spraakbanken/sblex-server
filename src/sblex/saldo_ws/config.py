from pydantic_settings import SettingsConfigDict

from sblex.sblex_server.settings import Settings
from sblex.telemetry.settings import OTelSettings


class SaldoWsSettings(Settings):
    model_config = SettingsConfigDict(
        env_file=".env", extra="ignore", env_nested_delimiter="__", env_prefix="SALDO_WS__"
    )


def read_settings_from_env() -> SaldoWsSettings:
    otel = OTelSettings(_env_prefix="SALDO_WS__")
    return SaldoWsSettings(otel=otel)
