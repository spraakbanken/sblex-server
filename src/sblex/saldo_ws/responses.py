from fastapi.responses import Response


class XMLResponse(Response):
    media_type = "application/xml"
