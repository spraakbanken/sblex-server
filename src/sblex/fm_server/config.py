from pydantic_settings import BaseSettings, SettingsConfigDict
from sblex.telemetry import OTelSettings


class Settings(BaseSettings):
    morphology_path: str
    otel: OTelSettings = OTelSettings()
    model_config = SettingsConfigDict(env_file=".env", extra="ignore")


def read_settings_from_env() -> Settings:
    otel = OTelSettings(_env_prefix="FM_SERVER_")
    settings = Settings(otel=otel)
    return settings
