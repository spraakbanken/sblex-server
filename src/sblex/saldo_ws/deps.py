from unittest import mock

import httpx
from fastapi import Depends, Request
from sblex.application.queries import FullformLexQuery, FullformQuery, LookupLid
from sblex.application.queries.inflection import InflectionTableQuery
from sblex.application.services import LookupService
from sblex.fm import Morphology
from sblex.fm.fm_runner import FmRunner
from sblex.infrastructure.queries import LookupFullformLexQuery
from sblex.infrastructure.queries.fm_runner_inflection import FmRunnerInflectionTable
from sblex.infrastructure.queries.http_morpology import HttpMorphology


def get_fm_client(request: Request) -> httpx.AsyncClient:
    return request.app.state._fm_client


def get_morphology(
    fm_client: httpx.AsyncClient = Depends(get_fm_client),  # noqa: B008
) -> Morphology:
    return HttpMorphology(http_client=fm_client)


def get_lookup_lid(request: Request) -> LookupLid:
    return request.app.state._lookup_lid


def get_fm_runner(request: Request) -> FmRunner:
    return request.app.state._fm_runner


def get_inflection_table_query(
    fm_runner: FmRunner = Depends(get_fm_runner),  # noqa: B008
) -> InflectionTableQuery:
    return FmRunnerInflectionTable(fm_runner=fm_runner)


def get_lookup_service(
    morphology: Morphology = Depends(get_morphology),  # noqa: B008
    lookup_lid: LookupLid = Depends(get_lookup_lid),  # noqa: B008
    inflection_table_query: InflectionTableQuery = Depends(get_inflection_table_query),  # noqa: B008
) -> LookupService:
    return LookupService(
        morphology=morphology, lookup_lid=lookup_lid, inflection_table=inflection_table_query
    )


def get_fullform_query() -> FullformQuery:
    query = mock.Mock(spec=FullformQuery)
    query.query = mock.Mock(return_value=b'{"c":"a"}')
    return query


def get_fullform_lex_query(
    lookup_service: LookupService = Depends(get_lookup_service),  # noqa: B008
) -> FullformLexQuery:
    return LookupFullformLexQuery(lookup_service=lookup_service)
