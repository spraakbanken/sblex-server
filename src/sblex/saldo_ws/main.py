from sblex.saldo_ws import config, server

settings = config.read_settings_from_env()
print(f"{settings=}")

app = server.create_saldo_ws_server(settings=settings)
