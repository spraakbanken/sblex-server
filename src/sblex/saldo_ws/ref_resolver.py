import urllib.error
import urllib.parse
import urllib.request

from fastapi import FastAPI

from sblex.formatting import prlex


def lexeme_ref(lids, *, app: FastAPI):
    if lids == "":
        return "*"
    else:
        return "+".join(
            [
                '<a href="{}">{}</a>'.format(
                    app.url_path_for("lids:lid-html", lid=urllib.parse.quote(lid)),
                    prlex(lid),
                )
                for lid in lids.split()
            ]
        )
