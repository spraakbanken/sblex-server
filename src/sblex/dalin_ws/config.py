from pydantic_settings import SettingsConfigDict
from sblex.fm_server.config import Settings
from sblex.telemetry.settings import OTelSettings


class DalinWsSettings(Settings):
    model_config = SettingsConfigDict(
        env_file=".env", extra="ignore", env_nested_delimiter="__", env_prefix="DALIN_WS__"
    )


def read_settings_from_env() -> DalinWsSettings:
    otel = OTelSettings(_env_prefix="DALIN_WS__")
    return DalinWsSettings(otel=otel)
