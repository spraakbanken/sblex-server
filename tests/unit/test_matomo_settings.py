import pytest
from sblex.saldo_ws.config import MatomoSettings


class TestMatomoSettings:
    def test_url_without_idsite_raises_value_error(self, snapshot) -> None:
        with pytest.raises(ValueError) as exc:
            MatomoSettings(matomo_url="http://example.com")
        assert str(exc) == snapshot
