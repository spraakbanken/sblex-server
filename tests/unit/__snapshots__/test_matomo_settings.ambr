# serializer version: 1
# name: TestMatomoSettings.test_url_without_idsite_raises_value_error
  '''
  1 validation error for MatomoSettings
    Value error, MATOMO_IDSITE must be an INT, got: 'None' (<class 'NoneType'>) [type=value_error, input_value={'matomo_url': 'http://example.com'}, input_type=dict]
      For further information visit https://errors.pydantic.dev/2.7/v/value_error
  '''
# ---
